src/main/java/com/duckblade/osrs/toa/features/ChestAudio.java

```diff
@@ -0,0 +1,181 @@
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.duckblade.osrs.toa.features;
+
+import com.duckblade.osrs.toa.TombsOfAmascutConfig;
+import com.duckblade.osrs.toa.module.PluginLifecycleComponent;
+import java.io.File;
+import java.io.IOException;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import javax.sound.sampled.AudioInputStream;
+import javax.sound.sampled.AudioSystem;
+import javax.sound.sampled.Clip;
+import javax.sound.sampled.FloatControl;
+import javax.sound.sampled.LineUnavailableException;
+import javax.sound.sampled.UnsupportedAudioFileException;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import net.runelite.api.DynamicObject;
+import net.runelite.api.GameObject;
+import net.runelite.api.events.GameObjectDespawned;
+import net.runelite.api.events.GameObjectSpawned;
+import net.runelite.api.events.GameTick;
+import net.runelite.client.RuneLite;
+import net.runelite.client.eventbus.Subscribe;
+import net.runelite.client.events.ConfigChanged;
+
+@Slf4j
+@Singleton
+@RequiredArgsConstructor(onConstructor_ = @Inject)
+public class ChestAudio implements PluginLifecycleComponent
+{
+	private static final int SARCOPHAGUS_OBJECT_ID = 46220;
+	private static final int SARCOPHAGUS_OBJECT_ID_2 = 46221;
+	private static final int OPENING_ANIMATION = 9505;
+
+	private Clip clip = null;
+	private GameObject chest =  null;
+
+	@Inject
+	private TombsOfAmascutConfig config;
+
+	@Override
+	public boolean isConfigEnabled(TombsOfAmascutConfig config)
+	{
+		return config.chestAudioEnable();
+	}
+
+	@Override
+	public void startUp()
+	{
+		// Nothing to do on startUp but its required for `PluginLifecycleComponent`
+	}
+
+	@Override
+	public void shutDown()
+	{
+		if (clip != null)
+		{
+			clip.close();
+		}
+		clip = null;
+	}
+
+	@Subscribe
+	public void onConfigChanged(ConfigChanged c)
+	{
+		if (!c.getGroup().equals(TombsOfAmascutConfig.CONFIG_GROUP))
+		{
+			return;
+		}
+
+		if (c.getKey().equals("chestAudioVolume"))
+		{
+			// Play clip when changing audio config so they can preview the change
+			playClip();
+		}
+	}
+
+	@Subscribe
+	private void onGameObjectSpawned(GameObjectSpawned e)
+	{
+		// Check both IDs as we still aren't sure which is correct for the purple.
+		if (e.getGameObject().getId() == SARCOPHAGUS_OBJECT_ID || e.getGameObject().getId() == SARCOPHAGUS_OBJECT_ID_2)
+		{
+			chest = e.getGameObject();
+		}
+	}
+
+	@Subscribe
+	private void onGameObjectDespawned(GameObjectDespawned e)
+	{
+		if (chest != null && chest.getId() == e.getGameObject().getId())
+		{
+			chest = null;
+		}
+	}
+
+	@Subscribe
+	private void onGameTick(GameTick e)
+	{
+		if (chest == null)
+		{
+			return;
+		}
+
+		if (((DynamicObject) chest.getRenderable()).getAnimation().getId() == OPENING_ANIMATION)
```

Comment:
`chest` can be of DynamicObject type instead of casting here, cast at assignment site if needed (better yet type check and log an error to be future-proof)

Reply:
`DynamicObject` does not have the `id` of the object so I'm not sure how I would rewrite the `onGameObjectDespawned` to account for the multiple IDs.